index/index.go

```diff
@@ -220,6 +221,53 @@ func (idx *Indexer) loopIteration(
 				doc:       doc,
 			}
 		case "document", "acl", "status":
+			query, err := json.Marshal(createLanguageQuery(item.Uuid, language))
+			if err != nil {
+				return 0, fmt.Errorf("marshal json: %w", err)
+			}
+
+			rootAlias := idx.getIndexTypeRoot(item.Type, "documents")
+
+			// Find all existing documents with the same Id but a different
+			// language
+			oldDocsResponse, err := idx.client.Search(
+				idx.client.Search.WithIndex(rootAlias),
+				idx.client.Search.WithBody(bytes.NewReader(query)),
+				idx.client.Search.WithContext(ctx),
+			)
+			if err != nil {
+				return 0, fmt.Errorf("lookup document: %w", err)
+			}
+
+			oldDocsBody, err := io.ReadAll(oldDocsResponse.Body)
```

Comment:
The body needs closing as well, and the case handling here has gotten quite long, so I would suggest factoring out to a separate function where defer can be used effectively for cleanup.

Reply:
Fixed.